src/logsumexp.jl

```diff
@@ -61,36 +61,59 @@ _logsumexp_onepass_reduce(X, ::Base.EltypeUnknown) = reduce(_logsumexp_onepass_o
 ## Reductions for one-pass algorithm: avoid expensive multiplications if numbers are reduced
 
 # reduce two numbers
-function _logsumexp_onepass_op(x1, x2)
-    a = x1 == x2 ? zero(x1 - x2) : -abs(x1 - x2)
-    xmax = x1 > x2 ? oftype(a, x1) : oftype(a, x2)
+function _logsumexp_onepass_op(x1::T, x2::T) where {T<:Number}
+    xmax, a = if x1 == x2
+        # handle `Inf` etc. correctly
+        x2, zero(x1 - x2)
+    elseif real(x1) > real(x2)
```

Comment:
I haven't thought too deeply about this, but how does it work to check only the real part? :-)

Reply:
The real part determines the absolute value of `exp(z)`, and hence it is used to determine which term is used to scale the sum of exponentials. We do not scale with the real part though but the full complex number since this allows us to use `log1p` for computing the logarithm of the scaled exponentials in the final reducing step, which avoids underflow.

Reply:
This is also explained in https://scicomp.stackexchange.com/questions/34273/log-sum-exp-trick-for-signed-complex-numbers#comment69041_34275.